### PR TITLE
Fixed bug for entering url manually. Prepared routing for incomplete profile

### LINKED ISSUE #86: [BUG] - Routing for manual URL entry goes to home page every time
## Description
If the user enters a URL manually or refreshes the page, it directs to home every time.

## Steps to Reproduce
Steps to reproduce the bug:
- [ ] Step 1 login
- [ ] Step 2 load any page (non-home)
- [ ] Step 3 refresh, or type in url manually.

## Expected Behavior
should direct to the user's entered url.

### PR DESCRIPTION
**Pull request by danieloh99**
Fixed bug for entering url manually. See Login.js. This required some changes to Profile.js, @ElwynCruz plz take a look. Basically, userInfo and userData were not playing nice together, since userInfo would not update when userData updated. It seems like userData doesn't always get updated right away, sometimes it starts out null, and then gets the data. But with the changes, everything works fine. #86 

Prepared routing for incomplete profile. The relevant code is in ProtectedRoute.js, but it is commented out for now, so that @jordanXYsoon1998 can enable it with the proper url once the corresponding Complete-Your-Profile page is ready. #76

We do not redirect to the EditProfile component (though I've written and commented out code that could do that if we want to), because technically, it has to direct to the Profile component and then tell it to open the EditProfile pop-up, which is somewhat messy.

### CODE REVIEW

File: src/components/Profile/Profile.js [hunk 1]
```diff
@@ -31,8 +30,27 @@ const useStyles = makeStyles((theme) => ({
 export default function Profile() {
   const classes = useStyles();
   const { userData } = useAuth();
```
Comment (ElwynCruz) [inline src/components/Profile/Profile.js:32]: so then userData updates whenever theres a change in the db, like if we edit the profile? does it make the page re-render with the new info without needing a refresh?

**Review by ElwynCruz — COMMENTED** [review]
looks good, only thing is I'm wary of just using the userData object since I'm not sure when it'll re-render with new information. If it already re-renders properly then it should be all good, but otherwise I think maintaining the state with the hook then updating that state as well as the db when making changes in the edit profile would be the way to go

File: src/components/Profile/Profile.js (hunk 1 above)
Comment (ElwynCruz) [reply src/components/Profile/Profile.js:32]: just tested it and looks good, should be good to merge

**Review by ElwynCruz — APPROVED** [review]

**Review by jordanXYsoon1998 — APPROVED** [review]
Code looks good to me. I should be able to take these changes and use them to complete my sprint code as well!

**Pull request merged**